BurnOutSharp/ProtectionType/Tages.cs

```diff
@@ -56,74 +55,132 @@ public string CheckContents(string file, byte[] fileContent, bool includeDebug,
+                    ./sys/VtPr.sys (a0acbc2f8e321e4f30c913c095e28af444058249)
+                    ./Wave.aif (SHA-1 is variable, file size is 81,920 bytes)
+                    ./Wave.alf (f82339d797be6da92f5d9dadeae9025385159057)
+                    ./Wave.apt (0351d0f3d4166362a1a9d838c9390a3d92945a44)
+                    ./Wave.axt (SHA-1 is variable, file size is 61,440 bytes)
+                */
+                new PathMatchSet(new List<string>
+                {
+                    Path.Combine("Sys", "Devx.sys").Replace("\\", "/"),
+                    Path.Combine("Sys", "VtPr.sys").Replace("\\", "/"),
+                    "Wave.aif",
+                    "Wave.alf",
+                    "Wave.apt",
+                    "Wave.axt",
+                }, "TAGES"),
 
-            if (protections.Count == 0)
-                return null;
-            else
-                return protections;
+                /*
+                    Currently only known to exist in "XIII" and (presumably) "Beyond Good & Evil".
+                */
+                new PathMatchSet(new List<string>
+                {
+                    "enodpl.sys",
+                    "ENODPL.VXD",
+                    "tandpl.sys",
+                    "TANDPL.VXD",
+                }, "TAGES"),
+
+                /*
+                    The directory of these files has been seen to be named two different things, with two different accompanying executables in the root of the directory.
+                    In the example where the directory is named "Drivers", the executable is named "Silent.exe".
+                    In the example where the directory is named "ELBdrivers", the executable is name "ELBDrivers.exe".
+
+                    Expected information about these checks (all of these are expected to be found together):
+                    ./9x/hwpsgt.vxd (40826e95f3ad8031b6debe15aca052c701288e04)
+                    ./9x/lemsgt.vxd (f82339d797be6da92f5d9dadeae9025385159057)
+                    ./NT/hwpsgt.sys (43f407ecdc0d87a3713126b757ccaad07ade285f)
+                    ./NT/lemsgt.sys (548dd6359abbcc8c84ce346d078664eeedc716f7)
+                    (The name and file size of the included executable vary, but there should always be one here.)
+                */
+                new PathMatchSet(new List<string>
+                {
+                    Path.Combine("9x", "hwpsgt.vxd").Replace("\\", "/"),
+                    Path.Combine("9x", "lemsgt.vxd").Replace("\\", "/"),
+                    Path.Combine("NT", "hwpsgt.sys").Replace("\\", "/"),
+                    Path.Combine("NT", "lemsgt.sys").Replace("\\", "/"),
+                }, "TAGES"),
+            };
+
+            return MatchUtil.GetAllMatches(files, matchers, any: true);
         }
 
         /// <inheritdoc/>
         public string CheckFilePath(string path)
         {
-            // "Wave.aif" was a former check here, but it over-matched
```

Comment:
Please at least keep some note about the various `wave.*` files that are associated.